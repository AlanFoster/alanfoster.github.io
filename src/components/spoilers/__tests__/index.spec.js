import React from "react";
import Component from "../";
import renderer from "react-test-renderer";
import * as reactstrap from "reactstrap";

reactstrap.Collapse = ({ isOpen, children }) => (isOpen ? children : null);

describe("spoilers", function() {
  describe("when closed", function() {
    it("does not render its children", function() {
      let tree;
      renderer.act(() => {
        tree = renderer.create(
          <Component>This text should *not* appear in the test</Component>
        );
      });

      expect(tree).toMatchSnapshot();
    });
  });

  describe("when open", function() {
    it("does not render its children", function() {
      let tree;
      renderer.act(() => {
        tree = renderer.create(
          <Component>This text should appear in the test</Component>
        );
      });

      renderer.act(() => {
        tree.root.findByProps({ className: "btn btn-link" }).props.onClick();
      });

      expect(tree).toMatchSnapshot();
    });
  });
});
