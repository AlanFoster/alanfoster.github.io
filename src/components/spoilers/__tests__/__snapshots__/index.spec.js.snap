// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`spoilers when closed does not render its children 1`] = `
<div>
  <div
    className="card"
  >
    <div
      className="card-header"
    >
      <h5
        className="card"
      >
        <button
          className="btn btn-link"
          onClick={[Function]}
        >
          View Spoiler?
        </button>
      </h5>
    </div>
  </div>
</div>
`;

exports[`spoilers when open does not render its children 1`] = `
<div>
  <div
    className="card"
  >
    <div
      className="card-header"
    >
      <h5
        className="card"
      >
        <button
          className="btn btn-link"
          onClick={[Function]}
        >
          View Spoiler?
        </button>
      </h5>
    </div>
    <div
      className="card-wrapper"
    >
      <div
        className="card-body"
      >
        This text should appear in the test
      </div>
    </div>
  </div>
</div>
`;
